src/lib/iptables.ts

```diff
@@ -1,35 +1,83 @@
-import * as Promise from 'bluebird';
+import * as Bluebird from 'bluebird';
 import * as childProcess from 'child_process';
+import * as _ from 'lodash';
+import { join } from 'path';
 
 // The following is exported so that we stub it in the tests
-export const execAsync = Promise.promisify(childProcess.exec);
 
-function clearIptablesRule(rule: string): Promise<void> {
-	return execAsync(`iptables -D ${rule}`).return();
+export const execAsync: (args: string) => Bluebird<string> = Bluebird.promisify(
+	childProcess.exec,
+);
+
+const getLoadedModules = _.once(async () => {
+	const modules = await execAsync('lsmod');
+	return _(modules)
+		.split('\n')
+		.map(modLine => modLine.split(' ')[0])
+		.reject(_.isNull)
+		.value();
+});
+
+export async function moduleLoaded(module: string): Promise<boolean> {
+	const modules = await getLoadedModules();
+	return _.find(modules, m => m === module) !== undefined;
+}
+
+export async function loadModules(): Promise<void> {
+	const moduleName = 'ip6_tables';
+	if (await moduleLoaded(moduleName)) {
+		console.log('ip6tables kernel module already loaded, skipping...');
+		return;
+	}
+	try {
+		// Get the kernel version so that we can load the module directly
+		const kernelVersion = (await execAsync('uname -r')).trim();
+		console.log(
+			`Loading ip6tables kernel module for kernel version ${kernelVersion}...`,
+		);
+		const modPath = join(
+			'/mnt/root/lib/modules',
+			kernelVersion,
+			'kernel/net/ipv6/netfilter/ip6_tables.ko',
+		);
+
+		await execAsync(`insmod ${modPath}`);
```

Comment:
This will work, as long as the ip6_tables module has no unmet dependencies, but too bad the busybox modprobe doesn't support `modprobe --dirname`... I wonder if it wouldn't be better to just symlink /lib/modules to /mnt/root/lib/modules inside the supervisor container?